server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -174,15 +174,76 @@ private Collection<DataSegment> doRetrieveUsedSegments(
   }
 
   @Override
-  public List<Pair<DataSegment, String>> retrieveUsedSegmentsAndCreatedDates(String dataSource)
+  public List<Pair<DataSegment, String>> retrieveUsedSegmentsAndCreatedDates(String dataSource, Interval interval)
   {
-    String rawQueryString = "SELECT created_date, payload FROM %1$s WHERE dataSource = :dataSource AND used = true";
-    final String queryString = StringUtils.format(rawQueryString, dbTables.getSegmentsTable());
+    StringBuilder queryBuilder = new StringBuilder(
+        "SELECT created_date, payload FROM %1$s WHERE dataSource = :dataSource AND used = true"
+    );
+
+    final boolean intervalStartIsEternityStart = Intervals.ETERNITY.getStart().equals(interval.getStart());
```

Comment:
Why do we need to do this special eternity handling? Are other methods in this class doing this too?

Reply:
This is important as such an interval may be passed to the method as an argument.
Other methods in this class do not implement it as they assume that no interval is passed when fetching segments for ETERNITY.